File: kotlinamesj/src/main/java/com/nagopy/android/kotlinames/property/required/KRequiredDoublePropertyName.java
package com.nagopy.android.kotlinames.property.required;

import com.nagopy.android.kotlinames.KRequiredPropertyName;
import com.nagopy.android.kotlinames.KSortablePropertyName;

public class KRequiredDoublePropertyName implements KRequiredPropertyName<Double>, KSortablePropertyName<Double> {

    private final String name;

    public KRequiredDoublePropertyName(String name) {
        this.name = name;
    }

    @Override
    public String name() {
        return name;
    }
}

File: kotlinames/src/main/kotlin/com/nagopy/android/kotlinames/property/required/KRequiredDoublePropertyName.kt
package com.nagopy.android.kotlinames.property.required

import com.nagopy.android.kotlinames.KRequiredPropertyName
import com.nagopy.android.kotlinames.KSortablePropertyName

class KRequiredDoublePropertyName(val name: String) : KRequiredPropertyName<Double>, KSortablePropertyName<Double> {

    override fun name(): String = name

}
